web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -163,10 +171,18 @@ def _finalist_office_hours_queryset(self):
                                               user_programs))
         return MentorProgramOfficeHour.objects.filter(
             reserved_by_user | unreserved & relevant_mentors,
-            start_date_time__range=[self.start_date, self.end_date])
+            start_date_time__range=[self.start_date, self.end_date]).annotate(
+                own_office_hour=Case(
+                    default=Value(False),
+                    output_field=BooleanField()))
 
     def _null_office_hours_queryset(self):
-        return MentorProgramOfficeHour.objects.none()
+        return MentorProgramOfficeHour.objects.none().annotate(
+                        own_office_hour=Case(
+                            When(mentor_id=self.target_user.id,
+                                 then=Value(True)),
+                            default=Value(False),
```

Comment:
Non blocking: `When(mentor_id=self.target_user.id, then=Value(True))` is not necessary for `none` objects